modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -1164,12 +1163,20 @@ private FltrLg createFilter(ArgFltr rf) {
 
         if (bnFrom == null || bnTo == null) {
             LOG.debug("jsonrpc - eth_newFilter(): from, to block parse failed");
-            return null;
+            filter.setFilterError("jsonrpc - eth_newFilter(): from, to block parse failed");
+            return filter;
         }
 
         if (bnTo != BEST_PENDING_BLOCK && bnFrom > bnTo) {
             LOG.debug("jsonrpc - eth_newFilter(): from block is after to block");
-            return null;
+            filter.setFilterError("jsonrpc - eth_newFilter(): from block is after to block");
+            return filter;
+        }
+
+        if (bnTo >= (bnFrom + BLOCKS_QUERY_MAX)) {
+            LOG.debug("jsonrpc - eth_newFilter(): can't query more than 1000 blocks");
+            filter.setFilterError("jsonrpc - eth_newFilter(): can't query more than 1000 blocks");
```

Comment:
Both messages above should use the constant value instead of `1000` so that the message stays consistent even when the limit changes.